### PR TITLE
HydraConnect refactor

### PR DESCRIPTION
**Pull request by ebenenglish**
So here's the refactor based on our discussions at Hydra Connect and on the Blacklight-development list. To summarize, the main changes are:

* add catalog#map view which displays all objects in Solr index with geospatial data
* catalog#index map view uses facet data to populate map, so all results are immediately viewable (rather than just current page's worth of items)
* maplet widget available for catalog#show view
* use GeoJSON to store coordinates with placenames in Solr
* marker popups display location name (or coordinates) and link to search params (rather than displaying results in the map view itself)
* marker clusters/markers display number of hits, rather than locations
* search control on map allows search for all items within current map window bounds
* map bounds and zoom are set dynamically when more than one location displayed on map
* add spatial search functionality to Blacklight
* render spatial constraints in catalog#index

Note: there are a handful of tests that are 'pending' because I couldn't get them passing despite extensive effort. But, they seemed important so I left them in -- hopefully someone can get these working?

Other note: I haven't updated the version. Let me know if that's something I should be doing before submitting the PR.

Thanks for everyone's patience on this!

### CODE REVIEW

File: blacklight-maps.gemspec [hunk 1]
```diff
@@ -23,6 +23,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency "leaflet-rails"
   spec.add_dependency "leaflet-markercluster-rails"
   spec.add_dependency "leaflet-sidebar-rails", "~> 0.0.2"
+  #spec.add_dependency "c_geohash", "~> 1.1.2" # don't need if location_rpt can't be faceted properly
```
Comment (mejackreed) [inline blacklight-maps.gemspec]: This seems like cruft here. Is it necessary?

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -34,5 +34,5 @@
 
 
 RSpec.configure do |config|
-
+  config.infer_spec_type_from_file_location!
```
Comment (mejackreed) [inline spec/spec_helper.rb]: This is causing the build to fail when using Rspec < 3.0 . Can we either check for the Rspec version, or update to using Rspec > 3?

Comment (mejackreed) [inline spec/spec_helper.rb]: Maybe in gemspec

```
gem 'rspec-rails', '~> 3.0'
```

Comment (ebenenglish) [inline spec/spec_helper.rb]: Seems fine.

File: blacklight-maps.gemspec (hunk 1 above)
Comment (ebenenglish) [inline blacklight-maps.gemspec]: Nope, leftover from some experimentation I was doing. Will remove.

**Comment by ebenenglish** [comment]
I pushed up a commit to address the gemspec/Rspec issues you commented on.

**Comment by mejackreed** [comment]
@ebenenglish great thanks. Also it looks like it will also need a rebase (merge conflicts). Can you please do a rebase off of master and then push back up?

**Comment by ebenenglish** [comment]
Rebase done; looks like build failing due to a few tests failing with Ruby 1.9.3. Are we still supporting this version of Ruby? If so I'll see if I can get these working...

**Comment by ebenenglish** [comment]
All tests should now be passing in Ruby 1.9.3. Latest build failed under 2.2.0 and 2.0.0 (but passed for 2.1.0 and 1.9.3) because of a weird Poltergeist timeout error:
```
Capybara::Poltergeist::TimeoutError:

Timed out waiting for response to {"name":"visit","args":["http://127.0.0.1:35506/catalog?q=korea\u0026view=maps"]}. It's possible that this happened because something took a very long time (for example a page load was slow). If so, setting the Poltergeist :timeout option to a higher value will help (see the docs for details). If increasing the timeout does not help, this is probably a bug in Poltergeist - please report it to the issue tracker.
```
Not sure if this is a random momentary glitch or what, but this didn't come up in previous builds, and all tests are passing on my end.

**Comment by mejackreed** [comment]
Ok removing jruby and 1.9.3 from travis with #57 ... rerunning travis to see if I can get it to pass.

**Comment by mejackreed** [comment]
@ebenenglish Do you have this deployed somewhere I can play with? Thanks!

**Comment by ebenenglish** [comment]
Yes, it's been deployed on the [Digital Commonwealth](http://www.digitalcommonwealth.org/) production site:
* ['Browse everything' map view](https://www.digitalcommonwealth.org/map)
* [Map view for search results](https://www.digitalcommonwealth.org/search?q=monuments&search_field=all_fields&utf8=%E2%9C%93&view=maps)
* [Maplet in catalog#show view](https://www.digitalcommonwealth.org/search/commonwealth:6d570460s)

**Pull request merged**